trino-aws-proxy-spi/src/main/java/io/trino/aws/proxy/spi/security/S3DatabaseSecurityFacadeProvider.java

```diff
@@ -18,6 +18,9 @@
 
 public interface S3DatabaseSecurityFacadeProvider
 {
+    // set as config value for "s3-security.type". Then bind a S3DatabaseSecurityFacadeProvider
+    String S3_DATABASE_SECURITY_IDENTIFIER = "s3DatabaseSecurity";
```

Comment:
Going back to our chat earlier, do we need to make this distinction anymore?

Could we create an `AbstractS3DatabaseSecurityFacade` that looks something like the below:
```java
public abstract class AbstractS3DatabaseSecurityFacade
        implements S3SecurityFacade
{
    @Override
    public SecurityResponse apply(Optional<String> lowercaseAction)
    {
        return tableName(lowercaseAction)
            .map(tableName -> tableOperation(tableName, lowercaseAction))
            .orElseGet(() -> nonTableOperation(lowercaseAction));
    }

    abstract Optional<String> tableName(Optional<String> lowercaseAction);

    SecurityResponse nonTableOperation(Optional<String> lowercaseAction)
    {
        return SUCCESS;
    }

    SecurityResponse tableOperation(String tableName, Optional<String> lowercaseAction)
    {
        return SUCCESS;
    }
}

```

Then all request security controllers are the same, and it's up to the facade implementer to decide how to implement it. They can inherit from `AbstractS3DatabaseSecurityFacade` if they wish to get the convenience of having 2 methods. But the rest of the proxy shouldn't really care IMHO.

Reply:
Yeah - good points. I'll come back to this after I've looked at the Identity changes.

Reply:
@vagaerg please see https://github.com/trinodb/aws-proxy/pull/128